### PR TITLE
Introduce new emission-factor-based bioenergy tax

### PR DESCRIPTION
**Pull request by merfort**
# Purpose of this PR

This PR introduces a new type of bioenergy tax. The new tax is based on an emission factor (EF) that can be read in via a switch, which is deactivated by default. The tax itself is calculated by multiplying the EF with the carbon price, resulting in a price markup per unit of bioenergy. The EF only informs the tax level, i.e. it does not affect emission fluxes. It can be read in on a regional level. A detailed description of this tax is found in the description of equation `q21_taxrevBio`.

In order to distinguish this new tax from the already existing bioenergy tax, the old tax is renamed from `cm_bioenergy_tax` to `cm_bioenergy_SustTax` as it is justified by general **sustainability** concerns, whereas the new EF-based tax only accounts for potential emissions that should be internalized.

Summary of changes:

- Introduction of (region-specific) EF-based tax
- Renaming of original bioenergy tax to bioenergy sustainability tax
- Removal of p21_tau_bioenergy_tax, which was only used for reporting (the reporting was adapted accordingly)
- Minor bug fix in magicc reporting
- Code clean-up

## Type of change

- [x] Bug fix 
- [x] Refactoring
- [x] New feature 
- [x] Minor change (default scenarios show no differences, since the switch is deactivated by default)

## Checklist:

- [x] My code follows the coding etiquette
- [x] I have performed a self-review of my own code
- [x] Changes are commented, particularly in hard-to-understand areas
- [x] I have updated the in-code documentation
- [x] I have adjusted reporting where it was needed
- [x] The model compiles and runs successfully (`Rscript start.R -q`)

## Further information (optional):

* Test runs are here: 
`/p/tmp/merfort/REMIND_development/remind/output`

* Comparison of results between different tax levels of the sustainability tax (`sustTax...`) and the EF tax (`EFTax...`), can be found in the following compare scenarios PDF: 
`/p/tmp/merfort/REMIND_development/remind/compScen-TEST_SustTax_EFTax_all-2022-10-12_10.26.47-H12.pdf`
It can be seen that taxing bioenergy according to and EF of 20 kgCO/GJ has (on a global average) a very similar effect as if the sustainability tax is set to 1.5 in terms of bioenergy demand reduction (compare the two greenish lines):
image

### CODE REVIEW

**Comment by merfort** [comment]
After pulling develop and test runs failed. Need to check why...

File: standalone/template.gms [hunk 1]
```diff
@@ -221,11 +222,16 @@ cm_CCS_chemicals       = 1;        !! def = 1
...
+cm_bioenergy_SustTax    = 1.5;            !! def = 1.5
+cm_bioenergy_EF_for_tax = 0;              !! def = 0
+$setGlobal cm_regi_bioenergy_EFTax  glob  !! def = glob
+cm_bioenergymaxscen     = 0;              !! def = 0
+cm_tradecost_bio        = 2;              !! def = 2
+$setglobal cm_LU_emi_scen  SSP2           !! def = SSP2
+cm_1stgen_phaseout      = 0;              !! def = 0
+$setglobal cm_tradbio_phaseout  default   !! def = default
+cm_biolc_tech_phaseout  = 0;              !! def = 0
+cm_cprice_red_factor    = 1;              !! def = 1
```
Comment (merfort) [inline standalone/template.gms:234]:
```suggestion
```

File: scripts/utils/gdx_rename.sh [hunk 2]
```diff
@@ -71,7 +71,7 @@ if (length(args) == 1) {
     c_iterative_target_adj = 'cm_iterative_target_adj', 
     c_co2_tax_2020         = 'cm_co2_tax_2020', 
     c_macscen              = 'c_macscen', 
-    c_bioenergy_tax        = 'cm_bioenergy_tax', 
+    c_bioenergy_tax        = 'cm_bioenergy_SustTax',
```
Comment (dklein-pik) [inline scripts/utils/gdx_rename.sh:74]: Shouldn't we keep this line and add a new one with:
`cm_bioenergy_tax = 'cm_bioenergy_SustTax',`

**Review by dklein-pik — CHANGES_REQUESTED** [review]

File: scripts/utils/gdx_rename.sh (hunk 2 above)
Comment (merfort) [inline scripts/utils/gdx_rename.sh:74]:
```suggestion
    c_bioenergy_tax        = 'cm_bioenergy_tax',
    cm_bioenergy_tax       = 'cm_bioenergy_SustTax',
```

Comment (merfort) [reply scripts/utils/gdx_rename.sh]: I changed it.

**Review by dklein-pik — APPROVED** [review]

File: scripts/utils/gdx_rename.sh [hunk 3]
```diff
@@ -71,7 +71,8 @@ if (length(args) == 1) {
     c_iterative_target_adj = 'cm_iterative_target_adj', 
     c_co2_tax_2020         = 'cm_co2_tax_2020', 
     c_macscen              = 'c_macscen', 
-    c_bioenergy_tax        = 'cm_bioenergy_tax', 
+    c_bioenergy_tax        = 'cm_bioenergy_tax',
```
Comment (merfort) [inline scripts/utils/gdx_rename.sh:74]:
```suggestion
    c_bioenergy_tax        = 'cm_bioenergy_SustTax',
```

**Pull request merged**